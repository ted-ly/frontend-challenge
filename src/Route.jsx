import React from 'react';

export default function Route({ path, children }) {
    const pathMatch = window.location.pathname === path;
    return pathMatch ? children : null;
}