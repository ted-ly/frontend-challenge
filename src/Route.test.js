import renderer from 'react-test-renderer';
import Route from './Route';

describe('Route', () => {
    it('should match snapshot', () => {
        const component = renderer.create(<Route path='/'><div>path</div></Route>);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('should match snapshot if route does not match', () => {
        global.window = Object.create(window);
        Object.defineProperty(window, 'location', {
            value: {
                pathname: '/something-else'
            },
            writable: true,
        })
        const component = renderer.create(<Route path='/'><div>path</div></Route>);
        expect(component.toJSON()).toMatchSnapshot();
    });
});