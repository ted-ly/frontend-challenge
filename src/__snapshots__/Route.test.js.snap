// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Route should match snapshot 1`] = `
<div>
  path
</div>
`;

exports[`Route should match snapshot if route does not match 1`] = `null`;
